src/create_map.py

```diff
@@ -46,10 +46,17 @@ def collect_Land_Use(roi, start_time, end_time):
     """
     Collect Land Use data to help predict land surface temperature based on land use, vegetation, and other features.
     """
-    land_use_data = ee.Image("MODIS/006/MCD12Q1").select("LC_Type1").clip(roi)
-    land_use_values = land_use_data.reduceRegion(
+    land_use_dataset = (
+        ee.ImageCollection("MODIS/006/MCD12Q1").filterDate(start_time, end_time).select("LC_Type1").clip(roi)
+    )
+    if not land_use_dataset:
+        raise Exception(f"Land use data for the year {start_time} is not available.")
```

Comment:
Similar to the run time error advice, but one step further: you can also define you own custom exceptions, see https://pybitesplatform.com/tips/defining-custom-exceptions